test/hpke_test.c

```diff
@@ -0,0 +1,1410 @@
+            memset(info, 'i', infolen);
+        } else {
+            infolen = 0;
+        }
+        if (hpke_mode == OSSL_HPKE_MODE_PSK
+            || hpke_mode == OSSL_HPKE_MODE_PSKAUTH) {
+            pskp = lpsk;
+            memset(lpsk, 'P', psklen);
+            pskidp = lpskid;
+            memset(lpskid, 'I', psklen - 1);
+            lpskid[psklen - 1] = '\0';
+        } else {
+            psklen = 0;
+        }
+        for (kemind = 0; /* iterate over the kems, kdfs and aeads */
+             overallresult == 1 &&
+             kemind != (sizeof(hpke_kem_list) / sizeof(uint16_t));
+             kemind++) {
+            uint16_t kem_id = hpke_kem_list[kemind];
+            size_t authpublen = OSSL_HPKE_MAXSIZE;
+            unsigned char authpub[OSSL_HPKE_MAXSIZE];
+            unsigned char *authpubp = NULL;
+            EVP_PKEY *authpriv_evp = NULL;
+
+            hpke_suite.kem_id = kem_id;
+            if ((hpke_mode == OSSL_HPKE_MODE_AUTH) ||
+                (hpke_mode == OSSL_HPKE_MODE_PSKAUTH)) {
+                if (TEST_true(OSSL_HPKE_keygen(hpke_suite, NULL, 0,
+                                               authpub, &authpublen,
+                                               &authpriv_evp,
+                                               testctx, NULL)) != 1) {
+                    overallresult = 0;
+                }
+                authpubp = authpub;
+            } else {
+                authpublen = 0;
+            }
+            for (kdfind = 0;
+                 overallresult == 1 &&
+                 kdfind != (sizeof(hpke_kdf_list) / sizeof(uint16_t));
+                 kdfind++) {
+                uint16_t kdf_id = hpke_kdf_list[kdfind];
+
+                hpke_suite.kdf_id = kdf_id;
+                for (aeadind = 0;
+                     overallresult == 1 &&
+                     aeadind != (sizeof(hpke_aead_list) / sizeof(uint16_t));
+                     aeadind++) {
+                    uint16_t aead_id = hpke_aead_list[aeadind];
+                    size_t publen = OSSL_HPKE_MAXSIZE;
+                    unsigned char pub[OSSL_HPKE_MAXSIZE];
+                    size_t senderpublen = OSSL_HPKE_MAXSIZE;
+                    unsigned char senderpub[OSSL_HPKE_MAXSIZE];
+                    size_t cipherlen = OSSL_HPKE_MAXSIZE;
+                    unsigned char cipher[OSSL_HPKE_MAXSIZE];
+                    size_t clearlen = OSSL_HPKE_MAXSIZE;
+                    unsigned char clear[OSSL_HPKE_MAXSIZE];
+
+                    hpke_suite.aead_id = aead_id;
+                    if (!TEST_true(OSSL_HPKE_keygen(hpke_suite, NULL, 0,
+                                                    pub, &publen, &privp,
+                                                    testctx, NULL)))
+                        overallresult = 0;
+                    ctx = OSSL_HPKE_CTX_new(hpke_mode, hpke_suite,
+                                            testctx, NULL);
+                    if (ctx == NULL) { overallresult = 0; }
```

Comment:
Did you intentionally leave out the TEST_int_eq().
If this thing fails it intermittently it would be hard to debug...

Reply:
fair enough - changed to use TEST_foo() variously